storey/writers.py

```diff
@@ -373,6 +373,13 @@ async def _emit(self, batch, batch_key, batch_time):
         if dir_path:
             self._file_system.makedirs(dir_path, exist_ok=True)
         file_path = self._path if self._single_file_mode else f'{dir_path}{uuid.uuid4()}.parquet'
+        # Remove nanosecs from timestamp
+        for name, _ in df.items():
+            if pd.core.dtypes.common.is_datetime64_dtype(df[name]):
+                df[name] = df[name].astype('datetime64[us]')
```

Comment:
Shouldn't we be doing this at the source? (DataframeSource)
At least for the time column.

Reply:
I placed it first there, but moved it here per OrZ's advice (not to lose data if it's not necessary).